smartsim/settings/launch_settings.py

```diff
@@ -165,6 +177,15 @@ def env_vars(self, value: dict[str, str | None]) -> None:
 
         :param value: The new environment mapping
         """
+        if not (
+            isinstance(value, dict)
+            and all(
+                isinstance(key, str) and isinstance(val, (str, type(None)))
+                for key, val in value.items()
+            )
+        ):
+            raise TypeError("env_vars argument was not of type dict of str and str")
```

Comment:
Ditto `dict[str, str | None]` vs `dict[str, str]` discrepancy
```suggestion
            and all(
                isinstance(key, str) and isinstance(val, (str, type(None)))
                for key, val in value.items()
            )
        ):
            raise TypeError("env_vars argument was not of type dict of str and str or None")
```